itests/kit/node_unmanaged.go

```diff
@@ -440,66 +379,228 @@ func (tm *TestUnmanagedMiner) waitForMutableDeadline(ctx context.Context, sector
-func (tm *TestUnmanagedMiner) OnboardCCSector(ctx context.Context, proofType abi.RegisteredSealProof) (abi.SectorNumber, chan WindowPostResp,
-	context.CancelFunc) {
-	req := require.New(tm.t)
+func (tm *TestUnmanagedMiner) NextSectorNumber() abi.SectorNumber {
 	sectorNumber := tm.currentSectorNum
 	tm.currentSectorNum++
+	return sectorNumber
+}
+
+func (tm *TestUnmanagedMiner) PrepareSectorForProveCommit(
+	ctx context.Context,
+	proofType abi.RegisteredSealProof,
+	sectorNumber abi.SectorNumber,
+	pieces []abi.PieceInfo,
+) (seedEpoch abi.ChainEpoch, proveCommit []byte) {
+
+	req := require.New(tm.t)
+
+	// Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
+	preCommitSealRandEpoch := tm.waitPreCommitSealRandomness(ctx, sectorNumber, proofType)
+
+	// Generate a Pre-Commit for the CC sector -> this persists the proof on the `TestUnmanagedMiner` Miner State
+	tm.generatePreCommit(ctx, sectorNumber, preCommitSealRandEpoch, proofType, pieces)
 
 	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
 
-	// Step 1: Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
-	preCommitSealRand := tm.waitPreCommitSealRandomness(ctx, sectorNumber)
+	if !proofType.IsNonInteractive() {
+		// Submit the Pre-Commit to the network
+		var uc *cid.Cid
+		if len(pieces) > 0 {
+			unsealedCid := tm.unsealedCids[sectorNumber]
+			uc = &unsealedCid
+		}
+		r, err := tm.SubmitMessage(ctx, &miner14.PreCommitSectorBatchParams2{
+			Sectors: []miner14.SectorPreCommitInfo{{
+				Expiration:    2880 * 300,
+				SectorNumber:  sectorNumber,
+				SealProof:     proofType,
+				SealedCID:     tm.sealedCids[sectorNumber],
+				SealRandEpoch: preCommitSealRandEpoch,
+				UnsealedCid:   uc,
+			}},
+		}, 1, builtin.MethodsMiner.PreCommitSectorBatch2)
+		req.NoError(err)
+		req.True(r.Receipt.ExitCode.IsSuccess())
+	}
+
+	// Generate a ProveCommit for the CC sector
+	var seedRandomness abi.InteractiveSealRandomness
+	seedEpoch, seedRandomness = tm.proveCommitWaitSeed(ctx, sectorNumber, proofType)
 
+	proveCommit = []byte{0xde, 0xad, 0xbe, 0xef} // mock prove commit
 	if !tm.mockProofs {
-		// Step 2: Write empty bytes that we want to seal i.e. create our CC sector
-		tm.makeAndSaveCCSector(ctx, sectorNumber)
+		proveCommit = tm.generateProveCommit(ctx, sectorNumber, proofType, seedRandomness, pieces)
+	}
+
+	return seedEpoch, proveCommit
+}
+
+func (tm *TestUnmanagedMiner) SubmitProveCommit(
+	ctx context.Context,
+	proofType abi.RegisteredSealProof,
+	sectorNumber abi.SectorNumber,
+	seedEpoch abi.ChainEpoch,
+	proveCommit []byte,
+	pieceManifest []miner14.PieceActivationManifest,
+) {
+
+	req := require.New(tm.t)
+
+	if proofType.IsNonInteractive() {
+		req.Nil(pieceManifest, "piece manifest should be nil for Non-interactive PoRep")
```

Comment:
We should also directly fail the call to `SnapDeal` then if proof type is NI.

Reply:
see https://github.com/filecoin-project/lotus/pull/12105, where we snap in a deal into an ni-porep sector

ni-porep is for CC only, no option to specify pieces or commd, but it's designed to onboard CC so you can snap deals into it later. One of the things covered in https://medium.com/@filoz/simplifying-the-onboarding-pipeline-f3b26ccbb758 is the ideal of SaaS CC sectors -> snap deals to "simplify" the data onboarding process. Perhaps this could end up being the primary data onboarding pathway into the future, even doing away with the need for PCS3